docs/source/gbd2019_models/hemoglobin_anemia_and_iron_deficiency/hemoglobin/index.rst

```diff
@@ -172,15 +147,63 @@ Data Description Tables
+**The following python code provides functions to accurately sample from the ensemble distribution with the hemoglobin-specific parameters defined above** 
+
+.. code-block:: python
+
+  import numpy as np
+  import pandas as pd
+  import scipy
+
+  def _gamma_ppf(propensity, mean, sd):
+  """Returns the quantile for the given quantile rank (`propensity`) of a Gamma
+  distribution with the specified mean and standard deviation.
+  """
+      shape = (mean / sd)**2
+      scale = sd**2 / mean
+      return scipy.stats.gamma(a=shape, scale=scale).ppf(propensity)
+
+  def _mirrored_gumbel_ppf(propensity, mean, sd):
+  """Returns the quantile for the given quantile rank (`propensity`) of a mirrored Gumbel
+  distribution with the specified mean and standard deviation.
+  """
+      x_max = 220 # data_values.HEMOGLOBIN_DISTRIBUTION.EXPOSURE_MAX
+      _alpha = x_max - mean - (sd * np.euler_gamma * np.sqrt(6) / np.pi)
+      scale = sd * np.sqrt(6) / np.pi
+      tmp = _alpha + (scale*np.euler_gamma)
+      alpha = _alpha + x_max - (2*tmp)
+      return scipy.stats.gumbel_r(alpha, scale=scale).ppf(propensity)
+
+  # Called `viv_calc_iron_nbs` in Kjell's notebook
+  def sample_from_hemoglobin_distribution(prop_dist, propensity, exposure_parameters):
+  """
+  Returns a sample from an ensemble distribution with the specified mean and
+  standard deviation (stored in `exposure_parameters`) that is 40% Gamma and
+  60% mirrored Gumbel. The sampled value is a function of the two propensities
+  `prop_dist` (used to choose whether to sample from the Gamma distribution or
+  the mirrored Gumbel distribution) and `propensity` (used as the quantile rank
+  for the selected distribution).
+  """
+      #propensity = clip(propensity)
+      exposure_data = exposure_parameters
+      mean = exposure_data['mean']
+      sd = exposure_data['sd']
+
+      gamma = prop_dist < 0.4
+      gumbel = ~gamma
+      ret_val = pd.Series(index=prop_dist.index, name='value')
+      ret_val.loc[gamma] = _gamma_ppf(propensity.loc[gamma], mean, sd)
+      ret_val.loc[gumbel] = _mirrored_gumbel_ppf(propensity.loc[gumbel], mean, sd)
+      return ret_val
 
 .. note::
 
-  This code has been translated to python in a notebook hosted `here <https://github.com/ihmeuw/vivarium_gates_lsff/blob/main/tests/lsff_iron_exposure.ipynb>`__.
+	While not explicitly enforced by the code above, all hemoglobin values should be non-zero positive numbers. The probability of sampling a negative value is small, but if it occurs, the value should be resampled until it is a positive number or clipped to a value of 1.
```

Comment:
If we clip negatives to 1.0, should we also clip positives that are less than 1.0 to this minimum value?

Reply:
I just found some citations that claim the lowest ever observed hemoglobin value in a patient that survived blood loss was 6 g/L and the lowest observed value in a stable patient not on critical care support was 14 (!!). I updated the strategy to clip anything less than 6 back to 6.